iot_config/src/gateway_service.rs

```diff
@@ -36,8 +41,13 @@ impl GatewayService {
         region_map: RegionMapReader,
         auth_cache: AuthCache,
     ) -> Result<Self> {
+        let gateway_cache = Arc::new(Cache::new());
+        let cache_clone = gateway_cache.clone();
+        tokio::spawn(async move { cache_clone.monitor(4, 0.25, CACHE_EVICTION_FREQUENCY).await });
```

Comment:
This needs to go into the run loop.. with something that can hopefully also tell it to shut down cleanly.. spawning in new has a bad smell

Reply:
I want to explore this more in our team call tomorrow; I personally have had bad experiences with spawned tasks in these grpc handlers not shutting down but others swear tokio handles this for them cleanly

Reply:
Sorry, I wasn't clear on the call earlier. Despite the fact that `Cache::monitor` doesn't complete (I said it doesn't return), it does yield. So it _could_ still make sense to to also select on a shutdown monitor to ensure this thread shuts down cleanly. I don't have a strong opinion on whether spawning is good or bad in the constructor however.